metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,303 @@
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=-1)
+@click.pass_obj
+def add(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.add_tag(list(tags))
+
+    obj.echo("Operation successful. New tags:", err=False)
+    obj.echo(print_tags(metaflow_run_client), err=False)
+
+
+@tag.command('remove',
+             help='Remove tags from a run.')
+@click.option('--run-id',
+              required=False,  # set False here so we can throw a better error message
+              default=None,
+              type=str,
+              help="Run ID of the specific run to tag.  [required]")
+@click.option('--namespace',
+              'user_namespace',
+              required=False,
+              default=None,
+              type=str,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=-1)
+@click.pass_obj
+def remove(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.remove_tag(list(tags))
+
+    obj.echo("Operation successful. New tags:")
+    obj.echo(print_tags(metaflow_run_client))
+
+@tag.command('replace',
+             help='Replace tags of a run.')
+@click.option('--run-id',
+              required=False,  # set False here so we can throw a better error message
+              default=None,
+              type=str,
+              help="Run ID of the specific run to tag.  [required]")
+@click.option('--namespace',
+              'user_namespace',
+              required=False,
+              default=None,
+              type=str,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=2)
+@click.pass_obj
+def replace(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.replace_tag([tags[0]], [tags[1]])
+
+    obj.echo("Operation successful. New tags:")
+    obj.echo(print_tags(metaflow_run_client))
+
+@tag.command('list',
+             help='List tags of a run.')
+@click.option('--run-id',
```

Comment:
we could make `run-id` an optional argument, so the command would be symmetric with existing commands:

`logs 1623618905573323/start`
`dump 1623618905573323/start`
`tag list 1623618905573323`